### ISSUE TITLE
Internal transition docs need some rewording

### ISSUE THREAD

**Issue by hnordt** [davidkpiano/xstate#275]
https://xstate.js.org/docs/guides/internal.html

The article starts showing that example:

```js
const wordMachine = Machine({
  key: 'word',
  initial: 'left',
  states: {
    left: {},
    right: {},
    center: {},
    justify: {}
  },
  on: {
    // internal transitions
    LEFT_CLICK: '.left',
    RIGHT_CLICK: '.right',
    CENTER_CLICK: '.center',
    JUSTIFY_CLICK: '.justify'
  }
});
```

Then states the following:

> The above machine will start in the 'direction.left' state, and based on what is clicked, will internally transition to its other child states.

To improve that I think we need to change the code sample to something like:

```js
const wordMachine = Machine({
  key: 'word',
  initial: "direction",
  states: {
    direction: {
      initial: 'left',
      states: {
        left: {},
        right: {},
        center: {},
        justify: {}
      },
      on: {
        LEFT_CLICK: "direction.left",
        RIGHT_CLICK: 'direction.right',
        CENTER_CLICK: 'direction.center',
        JUSTIFY_CLICK: 'direction.justify'
      }
    }
  }
});
```

And just then explain that `.left` is a shortcut for `<current_state_node>.left` (I hope I'm not wrong).

Also I think it's worth to explain that `direction.left` is short for `#<current_machine_key>.direction.left` (I hope I'm not wrong).

**Comment by davidkpiano** [comment]
Right, I'm going to redo this section with clearer examples.

**Comment by hnordt** [comment]
Closed by #344.
